### PR TITLE
[Clang] NFC: Move Arm type attributes to separate trailing object.

### PR DESCRIPTION
**Pull request by sdesmalen-arm**
This decouples the Arm type attributes from other bits, which means
the data will only be allocated when a function uses these Arm
attributes.

The first patch adds the bit `HasArmTypeAttributes` to
`FunctionTypeBitfields`, which grows from 62 bits to 63 bits.

In the second patch, I've moved this bit (`HasArmTypeAttributes`) to
`FunctionTypeExtraBitfields`, because it looks like the bits in
`FunctionTypeBitfields` are precious and we really don't want that struct
to grow beyond 64 bits.

I've split this out into two patches to explain the rationale, but those
can be squashed before merging.

### CODE REVIEW

**Review by erichkeane — APPROVED** [review]
This way of moving these further away is alright with me.  If Aaron is OK with it, than I am too.  So please wait for his approval too.

**Review by AaronBallman — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**
